### PR TITLE
1637 update documentation for release 0.5

### PR DESCRIPTION
**Pull request by prupipho**
## PR checklist

- Quick summary of changes
update documentation with latest changes
- Which issue does this PR relate to?
#1637

### CODE REVIEW

File: docs/contributing/application-architecture.md [hunk 1]
```diff
@@ -1,59 +1,67 @@
...
-  -  The front-end wizard (referred to as the [client](https://github.com/Microsoft/WebTemplateStudio/tree/dev/src/client)), written in [React](https://reactjs.org/) and [Typescript](https://www.typescriptlang.org/).
-  - Generation engine (referred to as [Core Template Studio](https://github.com/Microsoft/CoreTemplateStudio)) written in [.NET Core](https://dotnet.microsoft.com/download). 
-  
-![Architecture Diagram](../resources/webts-architecture-diagram.png)
+  1. The [extension's backend](#extension) (referred to as the [extension](../../src/extension)). Written in [Typescript](https://www.typescriptlang.org/).
+  1. The [frontend wizard](#client) (referred to as the [client](../../src/client)). Written in [React](https://reactjs.org/) and [Typescript](https://www.typescriptlang.org/).
+  1. The [generation engine](#Core-template-studio) (referred to as [Core Template Studio](https://github.com/Microsoft/CoreTemplateStudio)) written in [.NET Core](https://dotnet.microsoft.com/download). 
+
+
+<img alt="VSIX Download" src="../resources/webts-architecture-diagram.png" width="80%"  />
```
Comment (sibille) [inline docs/contributing/application-architecture.md]:
```suggestion
<img alt="Architecture Diagram" src="../resources/webts-architecture-diagram.png" width="80%"  />
```
This should be Architecture Diagram

File: docs/contributing/getting-started-developers.md [hunk 2]
```diff
@@ -9,103 +10,120 @@ This document covers:
...
-3. Install [Git](https://git-scm.com/downloads)
-4. Install [.NET Core SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1)
-5. Install [Visual Studio Code](https://code.visualstudio.com/)
-6. Install [Yarn](https://yarnpkg.com/en/docs/install). You will need to finish installing Node.js before you install Yarn.
-7. Install [Typescript](https://www.typescriptlang.org/) globally. `npm install -g typescript` 
-8. Install [React-scripts]() globally `npm install -g react-scripts`
-9. Run the command `npm config set scripts-prepend-node-path true`. This tells VSCode which Node version to run during the extension compilation (otherwise you'll get an error during the build process).
+1. Install [Gulp](https://gulpjs.com/)
+1. Install [Git](https://git-scm.com/downloads)
+1. Install [.NET Core SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1)
```
Comment (sibille) [inline docs/contributing/getting-started-developers.md:25]: Counter is showing 1 on all items

File: docs/contributing/getting-started-developers.md [hunk 3]
```diff
@@ -115,34 +133,38 @@ code --install-extension [extensionName].vsix
...
+To run the template tests go to the [`extension` folder](../../src/extension) and run `yarn template-tests` to run the tests.
 
 
 ## Under the Hood
 
 The following notes are inspired by the [vscode-webview-react](https://github.com/rebornix/vscode-webview-react)
 repository by [rebornix](https://github.com/rebornix):
 
-- We inline `index.html` content in `src/extension/src/extension.ts` when creating the webview
+- We inline `index.html` content in `ext-src/extension.ts`  when creating the webview
```
Comment (sibille) [inline docs/contributing/getting-started-developers.md]: Not sure if this path is correct

File: docs/generated-apps/services/azure-appservice.md [hunk 4]
```diff
@@ -1,50 +1,50 @@
...
 
 ## Creating App Service
-Once you hit generate on the summary page, Web Template Studio will create an Azure App Service that will be accessible from `http://<app_name>.azurewebsites.net`.
+Once you hit generate on the summary page, *Web Template Studio* will create an *Azure App Service* that will be accessible from `http://<app_name>.azurewebsites.net`.
 
-Web Template Studio uses an arm-template for Azure App Services (generated under the arm-templates directory). This template contains the definitions and parameters for all resources that need to deploy. Once Azure receives your template, it takes about 2-3 minutes to create the App Service and the App Service Plan. The app service initially contains an empty web app.
+*Web Template Studio* uses an arm-template for *Azure App Service*s (generated under the arm-templates directory). This template contains the definitions and parameters for all resources that need to deploy. Once Azure receives your template, it takes about 2-3 minutes to create the App Service and the App Service Plan. The app service initially contains an empty web app.
```
Comment (sibille) [inline docs/generated-apps/services/azure-appservice.md]:
```suggestion
*Web Template Studio* uses an arm-template for *Azure App Services* (generated under the arm-templates directory). This template contains the definitions and parameters for all resources that need to deploy. Once Azure receives your template, it takes about 2-3 minutes to create the App Service and the App Service Plan. The app service initially contains an empty web app.
```
typo

**Review by sibille — COMMENTED** [review]
Left a few comments for small changes

File: docs/generated-apps/services/azure-appservice.md [hunk 5]
```diff
@@ -1,50 +1,50 @@
 # Azure App Service
 
-Azure App Service is an HTTP-based service for hosting web applications, REST APIs, and mobile back ends. You can develop in your favorite language, be it .NET, .NET Core, Java, Ruby, Node.js, PHP, or Python. Applications run and scale with ease on both Windows and Linux-based environments. For more info about Azure App Service click [here](https://docs.microsoft.com/azure/app-service/overview).
+*Azure App Service* is an HTTP-based service for hosting web applications, REST APIs, and mobile backends. You can develop in your favorite language, be it .NET, .NET Core, Java, Ruby, Node.js, PHP, or Python. Applications run and scale with ease on both Windows and Linux-based environments. For more info about *Azure App Service* click [here](https://docs.microsoft.com/azure/app-service/overview).
 
-Web Template Studio offers you the functionality to create and deploy your application to Azure App Service from the wizard quickly and easly.
+*Web Template Studio* offers you the functionality to create and deploy your application to *Azure App Service* from the wizard quickly and easly.
```
Comment (sibille) [inline docs/generated-apps/services/azure-appservice.md]:
```suggestion
*Web Template Studio* offers you the functionality to create and deploy your application to *Azure App Service* from the wizard quickly and easily.
```

typo

File: docs/contributing/getting-started-developers.md (hunk 3 above)
Comment (prupipho) [reply docs/contributing/getting-started-developers.md]: Probably we can remove this "Under the Hood" section entirely. See what you think about it... I don´t see much value on it... If you check [here](https://github.com/rebornix/vscode-webview-react#under-the-hood) . It just seems to be this text copied, but doesn´t seem to apply much value on our documentation

File: docs/contributing/getting-started-developers.md (hunk 2 above)
Comment (prupipho) [reply docs/contributing/getting-started-developers.md:25]: This is my preferred way to use lists in markdown and seems that github is ok with it https://guides.github.com/features/mastering-markdown/ (search for **_Ordered_** on that page)
If you open the preview of the file, it works ok, but when you create the list you don´t bother about the numberof items and just put **1.** in all of them

Can change if you don´t like this approach :) I just find it easier to manage when you change/add/remove items from the list.

Comment (sibille) [reply docs/contributing/getting-started-developers.md:25]: Much easier! Wasn't aware of this possibility

File: docs/contributing/getting-started-developers.md [hunk 6]
```diff
@@ -115,34 +133,38 @@ code --install-extension [extensionName].vsix
...
+The **generate-tests** script generates all the different permutations of a Fullstack project (without azure services).
+Afterwards the **run-tests** script *installs the dependencies* for the generated project, runs *yarn start*, *yarn lint* and *yarn test* on all the generated projects.
 
-To run the template tests go to src/extension and run `yarn template-test` to run the tests.
+To run the template tests go to the [`extension` folder](../../src/extension) and run `yarn template-tests` to run the tests.
 
 
 ## Under the Hood
 
 The following notes are inspired by the [vscode-webview-react](https://github.com/rebornix/vscode-webview-react)
```
Comment (prupipho) [inline docs/contributing/getting-started-developers.md]:
```suggestion
The Architecture of this app was inspired by the [vscode-webview-react](https://github.com/rebornix/vscode-webview-react)

Find some notes about it below if you wish to get started with it:
```

File: docs/contributing/getting-started-developers.md (hunk 3 above)
Comment (prupipho) [inline docs/contributing/getting-started-developers.md]:
```suggestion
- We inline the initial html `<!DOCTYPE html> ... ` content in `src/extension/src/reactPanel.ts`  when creating the webview
```

**Review by sibille — APPROVED** [review]

**Pull request merged**